### PR TITLE
Expand session.data documentation

### PR DESCRIPTION
**Pull request by BrianJKoopman**
## Description

This is an attempt to expand on the documentation for `session.data`, describing how one should add information to their Agent Process `session.data` object and how to interact with that object in an OCS Client. To do so I've added the use of `session.data` to the Fake Data Agent.

## Motivation and Context

Using `session.data` is very helpful, but it's currently an underused feature across the existing Agents. Hoping an example in the Fake Data Agent will provide a good reference for Agent authors and encourage more use of this feature.

## How Has This Been Tested?

I've run the modified Fake Data Agent in a small test environment and run the example code shown.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

File: docs/agents/fake_data.rst [hunk 1]
```diff
@@ -14,6 +14,35 @@ Command-line / site config args
    :func: add_agent_args
    :prog: fake_data_agent.py
 
+Configuration File Examples
+---------------------------
+Below are configuration examples for the ocs config file and for running the Agent in a docker container.
+
+ocs-config
+``````````
+To configure the Fake Data Agent we need to add a FakeDAtaAgent block to our
```
Comment (mhasself) [inline docs/agents/fake_data.rst:23]: StrAy cap.

File: agents/fake_data/fake_data_agent.py [hunk 2]
```diff
@@ -44,6 +44,28 @@ def start_acq(self, session, params=None):
...
+                 "channel_04": 0.1019265554541543,
+                 "channel_05": 0.09389479275963578,
+                 "channel_06": 0.10071855402986646,
+                 "channel_07": 0.09601271802732826,
+                 "channel_08": 0.09760831143883832,
+                 "channel_09": 0.11345360178932645,
+                 "channel_10": 0.10047676575328081,
+                 "channel_11": 0.09534462609141414,
+                 "channel_12": 0.09654199950059912,
+                 "channel_13": 0.11051763608358373,
```
Comment (mhasself) [inline agents/fake_data/fake_data_agent.py:64]: It would probably be safe to leave out channels 02-13 here, and represent them with "..."...

**Review by mhasself — APPROVED** [review]
Thanks for the docs!

Is there somewhere to recommend that operation docstrings should include a description of what they put in session.data and what it means?

File: agents/fake_data/fake_data_agent.py [hunk 3]
```diff
@@ -44,6 +44,28 @@ def start_acq(self, session, params=None):
...
+                 "channel_14": 0.1062686192067794,
+                 "channel_15": 0.10793263271024509},
+             "last_updated":1600448753.9288929}
```
[the remark this answers is not in the document]
Comment (BrianJKoopman) [reply agents/fake_data/fake_data_agent.py:67]: Updated.

File: agents/fake_data/fake_data_agent.py (hunk 2 above)
Comment (BrianJKoopman) [reply agents/fake_data/fake_data_agent.py]: Since number of channels is sort of arbitrary I just trimmed it down to 4.

**Comment by BrianJKoopman** [comment]
> Is there somewhere to recommend that operation docstrings should include a description of what they put in session.data and what it means?

Tried to make that a little more clear in 16e2cf9 and also explained the values I'm putting in session.data for the fake data agent process.

We should maybe handle #163 first before merging this? We've both updated the fake data agent.

**Pull request merged**